[Feat] 모달 (로딩, 설문 링크 생성 완료, 녹음 종료, 가입 환영) 구현 & PR #122 모달 (다운로드, 삭제) 구현 포함

kyeoungwoon opened the pull request:
## #️⃣ Related Issues

> 연관된 모든 Issue를 작성해주세요.

- Issue #90 
- PR #122 의 변경사항 또한 포함되어 있습니다.

## 🧑‍💻 작업 내용

> 어떤 작업을 진행했는지 자세하게 작성해주세요.

- 로딩, 설문 링크 생성 완료, 녹음 종료 모달을 구현합니다.
- 설문 링크 생성 완료 모달에서는 복사 버튼에 임의로 토스트를 추가하였습니다. (alert 띄우기에는 테스트 할 때 번거로웠던 관계로)
  - 해당 부분 임의로 생성한 토스트이기에 추가로 hook 등으로 생성하면 좋을 것 같아 이슈 생성해두었습니다. #124 

## 💾 작업 결과 (선택)

> 사진, 영상 등을 첨부해주세요.

- storybook은 추후에 작업 예정입니다. 사진 첨부드립니다.

<img width="879" height="601" alt="image" src="https://github.com/user-attachments/assets/6b77afde-bb90-4db9-a476-ef982b423cb3" />
<img width="872" height="776" alt="image" src="https://github.com/user-attachments/assets/dec11c82-d98b-4fa9-871e-66cbfec837ab" />


## 💬 리뷰 시 요청사항 (선택)

> Reviewer는 코드 리뷰의 코멘트에 코멘트를 강조하고 싶은 정도를 [Pn 규칙](https://blog.banksalad.com/tech/banksalad-code-review-culture/#%EC%BB%A4%EB%AE%A4%EB%8B%88%EC%BC%80%EC%9D%B4%EC%85%98-%EB%B9%84%EC%9A%A9%EC%9D%84-%EC%A4%84%EC%9D%B4%EA%B8%B0-%EC%9C%84%ED%95%9C-pn-%EB%A3%B0)에
> 맞춰서 표기해 주세요.

- 단계별로 PR을 생성했다가, 새로운 기능과 함께 작업하며 이전 PR close 처리를 병행하고 있습니다.
- 히스토리는 계속 기록중이니, 각 PR 및 Issue 참고해서 리뷰해주세요. (커밋별로 보시면서 review 해주셔도 좋을 것 같아요.)
- 최종 용량이 좀 클 예정입니다 ㅎㅎ;

## 📝 Checklist

- [ ] Reviewer를 추가했나요?
- [ ] Convention을 준수했나요?

duwlsssss committed:
✨ Feat: 로딩 바 컴포넌트 제작

duwlsssss committed:
✨ Feat: ProgressBar 추가

duwlsssss committed:
💄 Ui: 디자인 시스템 반영

duwlsssss committed:
🐛 Bug: 로딩바 스토리 이름 중복 수정

duwlsssss committed:
♻️ Refactor: framer-motion 적용

duwlsssss committed:
🔥 Remove: framer-motion 제거

- framer-motion의 크기가 커, 간단한 애니메이션에 적용하기 위해 사용하는 건 부적절하다 생각해 제거함

duwlsssss committed:
🐛 Bug: 스토리북에 globals.css 반영

duwlsssss committed:
✏️ Typo: 서버 컴포넌트 server suffix 붙이기

duwlsssss committed:
🔀 Merge: develop merge

duwlsssss committed:
♻️ Refactor: pxr 사용

kyeoungwoon committed:
✨ Feat: 모달 (삭제) 구현

storybook 또한 적용하였습니다.

kyeoungwoon committed:
🩹 Chore: 사용하지 않는 import 제거

그러합니다

kyeoungwoon committed:
🔧 Fix: 원형 SVG 아이콘 컴포넌트에 rounded 추가

해당 아이콘을 사용하는 버튼에 hover 시 속성을 설정했는데 사각형으로 공백이 생겨 이를 해결

kyeoungwoon committed:
✨ Feat: X 아이콘에 상태 추가

그러합니다

kyeoungwoon committed:
✨ Feat: 모달 (다운로드) 에서 사용하는 버튼 컴포넌트화

디자인 시스템에 존재하지 않았던 부분입니다

kyeoungwoon committed:
✨ Feat: 다운로드 모달

모달 (삭제)는 이름 변경 때문에 추적되는 것으로 보입니다

kyeoungwoon committed:
🩹 Chore: modal test page

모달 제작 중 테스트용으로 활용한 페이지입니다

kyeoungwoon committed:
✨ Feat: LoadingModal 생성  및 test page 정리

그러합니다

kyeoungwoon committed:
🔀 Merge: 로딩 바 컴포넌트 작업본 merge

여진이가 작업한 컴포넌트가 아직 develop에 merge 전이라서 직접 해당 브랜치를 가져왔습니다.

kyeoungwoon committed:
♻️ Refactor: 파일 다운로드 모달 분리

LoadingBar를 사용하는 것과, ProgressBar를 사용하는 점이 달라 파일을 구분하였습니다.

kyeoungwoon committed:
✨ Feat: 모달 (녹음 종료) 구현

그러합니다!

kyeoungwoon committed:
✨ Feat: 모달(설문 링크 생성 완료) 제작

복사 버튼 클릭 시 임의로 토스트 띄우도록 함.

kyeoungwoon committed:
✨ Feat: 모달 (가입 환영) 제작

파일명에 누락된 server prefix도 추가했습니다.

kyeoungwoon committed:
🔧 Fix: 링크 표시 텍스트 중앙정렬에서 좌측정렬로 변경

디자이너님 요청사항 반영

kyeoungwoon committed:
🔧 Fix: 가입 환영 모달 내 padding 값 변경

디자이너님 요청 사항

S-Gihun commented on src/common/components/modals/DeleteModal/DeleteModal.types.ts:
P1: 전체적으로 types 파일에 "use client" 제거가 필요해보입니다.

kyeoungwoon replied:
같은 파일에 작성 후에 IDE 기능 이용해서 분리하다보니 자동으로 붙은걸 제거를 안 했었네요.
수정처리 하겠습니다. (필요 없는 것 맞겠죠?)

S-Gihun replied:
네 types 폴더에는 제거하는 것이 맞는 거 같습니다.

kyeoungwoon committed:
🔀 Merge: resolve conflict with develop

shadow.css와 shadows.css가 모두 존재하여 shadows.css로 통합처리 하였습니다.

S-Gihun commented on src/common/components/modals/DeleteModal/DeleteModal.client.tsx:
```diff
@@ -0,0 +1,78 @@
+    },
+  };
+
+  let abortButton = <CancelButton onClick={onAbort} buttonType={CancelButtonType.SIZE_38} />;
```
P4: 아래와 같이 작성하면, 가독성도 높아지고 break 만 뱉어내는 case 도 없앨 수 있어서 한 번 봐보시면 좋을 거 같아요

```ts
const getModalButtons = (
  modalType: DeleteModalType,
  onAbort: () => void,
  onProceed: () => void,
): [React.ReactNode, React.ReactNode] => {
  switch (modalType) {
    case DeleteModalType.LEAVE_MEETING_RECORD:
      return [
        <CancelButton onClick={onAbort} buttonType={CancelButtonType.SIZE_38} />,
        <MoveButton onClick={onProceed} />,
      ];
    case DeleteModalType.CONNECT_INSTAGRAM_ACCOUNT:
      return [
        <SkipForNowButton onClick={onAbort} buttonType={SkipForNowButtonType.SIZE_38} />,
        <AccountConnectButton onClick={onProceed} />,
      ];
    default:
      return [
        <CancelButton onClick={onAbort} buttonType={CancelButtonType.SIZE_38} />,
        <DeleteButton onClick={onProceed} />,
      ];
  }
};
```

```ts
const DeleteModal = ({ modalType, onAbort, onProceed }: DeleteModalProps) => {
  const [abortButton, proceedButton] = getModalButtons(modalType, onAbort, onProceed);
  const { main, sub } = modalTexts[modalType];

  return (
    <div className="w-398pxr h-250pxr rounded-12pxr shadow-modal flex flex-col items-center justify-center">
      <IndividualIcons state={IndividualIconsState.WARNING} />

      <div className="space-y-6pxr mt-13pxr w-317pxr flex flex-col items-center justify-center text-center">
        <p className="text-t3-bd text-black">{main}</p>
        <p className="text-b3-rg text-gray-300">{sub}</p>
      </div>

      <div className="space-x-10pxr mt-28pxr flex flex-row">
        {abortButton}
        {proceedButton}
      </div>
    </div>
  );
};
```

kyeoungwoon replied:
비슷한 방법으로 수정해두었습니다!

코드 쓰다가 잠깐 나갔다 와서 다시 써서 그런가 이상하게 써놨네요. 😢

S-Gihun commented on src/common/components/modals/LoadingModal/LoadingModal.client.tsx:
```diff
@@ -0,0 +1,42 @@
+  };
+
+  return (
+    <div className="w-396pxr h-240pxr rounded-16pxr shadow-modal relative flex flex-col items-center justify-center">
```
P4: 해당 div 가 계속 반복 되고 있어서 BaseModalContainer 같은 걸 만들어서 관리하면 중복 코드를 줄일 수 있을 거 같습니다 : )

kyeoungwoon replied:
P5: 요건 defaultmodal 파일 있는걸 보시면 알겠지만 추후 onClose의 default action과 더불어서 설정 예정이였습니다! (아직 공부 중 ..)

S-Gihun reviewed (commented):
P1: Vercel과 Storybook CI/CD 문제 해결 후 develop 에 Merge 부탁드립니다.

kyeoungwoon committed:
🔧 Fix: delete 'use client' type files

IDE가 그랬어요

kyeoungwoon committed:
♻️ Refactor: 불필요한 조건문들 정리

deletemodal에 존재하던 불필요한 부분 제거

kyeoungwoon committed:
🐛 Bug: CI/CD 실패하는 것 수정

모노레포 방식을 사용할 때, pnpm-workspace.yaml을 활용해서
패키지를 관리할 수 있습니다.

@evilmartians/lefthook, @sentry/cli, @tailwindcss/oxide, core-js-pure, esbuild, sharp, unrs-resolver
해당 패키지는 pnpm approve-builds 를 통해서 빌드 스크립트를 실행할 수 있도록
허용해주어야 하는데, 해당 허용 과정에서 pnpm이 자동 생성한 파일이
pnpm-workspace.yaml이고, 그 안에 필수로 필요한 packages 옵션이 없어서
CI/CD가 실패하는 문제가 발생했었습니다. 그를 해결하였습니다.

duwlsssss reviewed (approved):
P5: 고생하셨습니다!

S-Gihun reviewed (approved):
P5: 고생하셨습니다 :)
